pystac_client/stac_api_io.py

```diff
@@ -50,33 +50,23 @@ def __init__(
         # TODO - this should super() to parent class
         self.session = Session()
         self.session.headers.update(headers or {})
-        self.session.params.update(parameters or {})
+
+        if hasattr(self.session.params, "update"):
```

Comment:
How could the `Session` not have a `dict` as its `params` attribute? Is it a `requests` version thing?

Reply:
yeah, params is typed as the completely incomprehensible union of several nested layers of type hints. The other option is to ignore this, which is probleably more reasonable to do